source/fuzz/transformation_inline_function.cpp

```diff
@@ -0,0 +1,239 @@
+                 cloned_label_instruction->result_id()) &&
+             "Label id must be mapped to a fresh id.");
+      uint32_t fresh_id =
+          message_.result_id_map().at(cloned_label_instruction->result_id());
+      cloned_label_instruction->SetResultId(fresh_id);
+      function_call_instruction->InsertBefore(
+          std::unique_ptr<opt::Instruction>(cloned_label_instruction));
+      fuzzerutil::UpdateModuleIdBound(ir_context, fresh_id);
+    }
+
+    for (auto& instruction : block) {
+      // Replaces the operand ids with their mapped result ids.
+      auto cloned_instruction = instruction.Clone(ir_context);
+      cloned_instruction->ForEachInId(
+          [this, called_function, function_call_instruction](uint32_t* id) {
+            // If the id is mapped, then set it to its mapped value.
+            if (message_.result_id_map().count(*id)) {
+              *id = message_.result_id_map().at(*id);
+              return;
+            }
+
+            uint32_t parameter_index = 0;
+            called_function->ForEachParam(
+                [id, function_call_instruction,
+                 &parameter_index](opt::Instruction* parameter_instruction) {
+                  // If the id is a function parameter, then set it to the
+                  // parameter value passed in the function call instruction.
+                  if (*id == parameter_instruction->result_id()) {
+                    *id = function_call_instruction->GetSingleWordInOperand(
+                        parameter_index + 1);
+                  }
+                  parameter_index++;
+                });
+          });
+
+      // If |cloned_instruction| has a result id, then set it to its mapped
+      // value.
+      if (cloned_instruction->HasResultId() &&
+          message_.result_id_map().count(cloned_instruction->result_id())) {
+        assert(message_.result_id_map().contains(
+                   cloned_instruction->result_id()) &&
+               "Result id must be mapped to a fresh id.");
+        uint32_t result_id =
+            message_.result_id_map().at(cloned_instruction->result_id());
+        cloned_instruction->SetResultId(result_id);
+        fuzzerutil::UpdateModuleIdBound(ir_context, result_id);
+      }
+
+      // The return instruction will be changed into an OpBranch to the basic
+      // block that follows the block containing the function call.
+      if (spvOpcodeIsReturn(cloned_instruction->opcode())) {
+        uint32_t following_block_id =
+            ir_context->get_instr_block(function_call_instruction)
+                ->terminator()
+                ->GetSingleWordInOperand(0);
+        switch (cloned_instruction->opcode()) {
+          case SpvOpReturn:
+            cloned_instruction->AddOperand(
+                {SPV_OPERAND_TYPE_ID, {following_block_id}});
+            break;
+          case SpvOpReturnValue: {
+            auto returned_value_id =
+                GetReturnValueId(ir_context, called_function);
```

Comment:
Could you just get the returned value id from `instruction`? You could also rename it `original_instruction` for clarity and less chance of name clashes.

Reply:
You're right.
Done.